builtin/v8/miner/bitfield_queue.go

```diff
@@ -0,0 +1,160 @@
+package miner
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/filecoin-project/go-state-types/builtin"
+	"github.com/filecoin-project/go-state-types/builtin/v8/util/adt"
+
+	"github.com/filecoin-project/go-bitfield"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+// Wrapper for working with an AMT[ChainEpoch]*Bitfield functioning as a queue, bucketed by epoch.
+// Keys in the queue are quantized (upwards), modulo some offset, to reduce the cardinality of keys.
+type BitfieldQueue struct {
+	*adt.Array
+	quant builtin.QuantSpec
+}
+
+func LoadBitfieldQueue(store adt.Store, root cid.Cid, quant builtin.QuantSpec, bitwidth int) (BitfieldQueue, error) {
+	arr, err := adt.AsArray(store, root, bitwidth)
+	if err != nil {
+		return BitfieldQueue{}, xerrors.Errorf("failed to load epoch queue %v: %w", root, err)
+	}
+	return BitfieldQueue{arr, quant}, nil
+}
+
+// Adds values to the queue entry for an epoch.
+func (q BitfieldQueue) AddToQueue(rawEpoch abi.ChainEpoch, values bitfield.BitField) error {
```

Comment:
These are an interesting case. There is logic here tightly coupled to the actor implementations. You've taken all of the mutation methods as well as inspection. It would be really annoying for future Go actor development if this logic were separate from the main actor logic like this – so it's a good think we're not doing any!

One possible justification for including this here is for future migrations (but the long term plan for that should be that they're written in Rust).

For now, I think I would suggest excluding all the mutation methods, and keeping this repo as thin as possible. Is there a case I'm missing that motivates including them?